python-threatexchange/threatexchange/threat_updates.py

```diff
@@ -0,0 +1,375 @@
+            new_deltas.append(
+                ThreatUpdatesDelta(
+                    self.privacy_group,
+                    prev.end,
+                )
+            )
+        prev.end = end
+
+        return self, new_deltas
+
+    def incremental_sync_from_threatexchange(
+        self,
+        api: ThreatExchangeAPI,
+        *,
+        limit: int = 0,
+        progress_fn=lambda x: None,
+    ) -> None:
+        """
+        Fetch from threat_updates to get a more up-to-date copy of the data.
+        """
+
+        # TODO actually implement fancy threading logic
+        # alternative - instead make the API give hints about where to start
+        # fetches, which will mean that fancy threading logic will be simpler
+        while not self.done:
+            for update in self.one_fetch(api):
+                progress_fn(update)
+                if limit:
+                    limit -= 1
+                    if limit <= 0:
+                        return
+
+
+class ThreatUpdateCheckpoint(t.NamedTuple):
+    """
+    State about the progress of a /threat_updates-backed state.
+
+    If a client does not resume tailing the threat_updates endpoint fast enough,
+    deletion records will be removed, making it impossible to determine which
+    records should be retained without refetching the entire dataset from scratch.
+    The API implementation will retain for 90 days:
+    https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-updates/
+    """
+
+    # See docstring about tailing fast enough
+    DEFAULT_REFETCH_SEC = 3600 * 24 * 85  # 85 days
+
+    # When was the last time we started or the furthest we've seen,
+    # to check against the store getting too stale
+    last_fetch_time: int = 0
+    # Where should we resume from?
+    fetch_checkpoint: int = 0
+
+    def get_updated(self, delta: ThreatUpdatesDelta) -> "ThreatUpdateCheckpoint":
+        return ThreatUpdateCheckpoint(
+            max(self.last_fetch_time, delta.current),
+            max(self.fetch_checkpoint, delta.current),
+        )
+
+    @property
+    def stale(self):
+        """Is this checkpoint so old as to be invalid?"""
+        return self.last_fetch_time + self.DEFAULT_REFETCH_SEC < time.time()
+
+
+class ThreatUpdatesStore:
+    """
+    A wrapper for ThreatIndicator records for a single Collaboration
+
+    There is a unique file for each combination of:
+     * IndicatorType
+     * PrivacyGroup
+
+    The contents of file does not strip anything from the API
+    response, so can potentially contain a lot of data.
+    """
+
+    def __init__(
+        self,
+        privacy_group: int,
+    ) -> None:
+        self.privacy_group = privacy_group
+        self.checkpoint = None
+
+    @property
+    def fetch_checkpoint(self):
+        return self.checkpoint.fetch_checkpoint
+
+    def reset(self) -> None:
+        """Toss old state and begin anew"""
+        self.checkpoint = ThreatUpdateCheckpoint(time.time(), 0)
```

Comment:
nit: why do we set `last_fetch_time` to `time.time()` here? 
shouldn't we wait for at least one successful fetch before then?

Reply:
Thought it made most sense here, but can move it somewhere else.